sentry_sdk/integrations/quart.py

```diff
@@ -0,0 +1,173 @@
+    # type: (Quart, **Any) -> None
+    hub = Hub.current
+    integration = hub.get_integration(QuartIntegration)
+    if integration is None:
+        return
+
+    app = _app_ctx_stack.top.app
+    with hub.configure_scope() as scope:
+        if _request_ctx_stack.top is not None:
+            request_websocket = _request_ctx_stack.top.request
+        if _websocket_ctx_stack.top is not None:
+            request_websocket = _websocket_ctx_stack.top.websocket
+
+        # Set the transaction name here, but rely on ASGI middleware
+        # to actually start the transaction
+        try:
+            if integration.transaction_style == "endpoint":
+                scope.transaction = request_websocket.url_rule.endpoint
+            elif integration.transaction_style == "url":
+                scope.transaction = request_websocket.url_rule.rule
+        except Exception:
+            pass
+
+        evt_processor = _make_request_event_processor(
+            app, request_websocket, integration
+        )
+        scope.add_event_processor(evt_processor)
+
+
+def _make_request_event_processor(app, request, integration):
+    # type: (Quart, Request, QuartIntegration) -> EventProcessor
+    def inner(event, hint):
+        # type: (Dict[str, Any], Dict[str, Any]) -> Dict[str, Any]
+        # if the request is gone we are fine not logging the data from
+        # it.  This might happen if the processor is pushed away to
+        # another thread.
+        if request is None:
+            return event
+
+        with capture_internal_exceptions():
+            # TODO: Figure out what to do with request body. Methods on request
+            # are async, but event processors are not.
+
+            request_info = event.setdefault("request", {})
+            request_info["url"] = request.url
+            request_info["query_string"] = request.query_string
+            request_info["method"] = request.method
+            request_info["headers"] = _filter_headers(dict(request.headers))
+
+            if _should_send_default_pii():
+                request_info["env"] = {"REMOTE_ADDR": request.access_route[0]}
+                _add_user_to_event(event)
+
+        return event
+
+    return inner
+
+
+def _capture_exception(sender, exception, **kwargs):
+    # type: (Quart, Union[ValueError, BaseException], **Any) -> None
+    hub = Hub.current
+    if hub.get_integration(QuartIntegration) is None:
+        return
+
+    # If an integration is there, a client has to be there.
+    client = hub.client  # type: Any
+
+    event, hint = event_from_exception(
+        exception,
+        client_options=client.options,
+        mechanism={"type": "quart", "handled": False},
+    )
+
+    hub.capture_event(event, hint=hint)
+
+
+def _add_user_to_event(event):
+    # type: (Dict[str, Any]) -> None
+    if quart_auth is None:
+        return
+
+    print("!!!!!!!!!!!!!!!!!!!!!", quart_auth.current_user._auth_id)
```

Comment:
This might be a leftover debug print statement

Reply:
Woops, thanks.